jwkutil/validate.go

```diff
@@ -0,0 +1,86 @@
+package jwkutil
+
+import (
+	"errors"
+	"fmt"
+
+	"slices"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+var (
+	ValidRSAAlgorithms   = []jwa.SignatureAlgorithm{jwa.PS256, jwa.PS384, jwa.PS512}
+	ValidECAlgorithms    = []jwa.SignatureAlgorithm{jwa.ES256, jwa.ES384, jwa.ES512}
+	ValidOctetAlgorithms = []jwa.SignatureAlgorithm{jwa.HS256, jwa.HS384, jwa.HS512}
+	ValidOKPAlgorithms   = []jwa.SignatureAlgorithm{jwa.EdDSA}
+
+	ValidSigningAlgorithms = concat(
+		ValidOctetAlgorithms,
+		ValidRSAAlgorithms,
+		ValidECAlgorithms,
+		ValidOKPAlgorithms,
+	)
+)
+
+var (
+	ErrKeyMissingAlg             = errors.New("key is missing algorithm")
+	ErrUnsupportedAlgForKeyType  = errors.New("unsupported key type")
+	ErrInvalidSigningAlgorithm   = errors.New("invalid signing algorithm")
+	ErrUnsupportSigningAlgorithm = errors.New("unsupported signing algorithm for key type")
+)
+
+func Validate(key jwk.Key) error {
```

Comment:
hmm, this one was missed? Validate seems deserving!

Reply:
Oh, missed upstream in agent.

Reply:
i'll get something in there :)